### PR TITLE
Remove aggregation fields for non-List relationships

### LINKED ISSUE #2969: Aggregate filter is wrongly generated for non-List relationship
### Type definitions

```GraphQL
type Post {
    content: String!
    author: User! @relationship(type: "HAS_AUTHOR", direction: OUT)
}

type User {
    id: ID!
    name: String!
    posts: [Post!]! @relationship(type: "HAS_AUTHOR", direction: IN)
}
```

### Test data

_No response_

### Steps to reproduce

1. Generate a schema using the above type definitions

### What happened

The following input type will be generated:

```gql
input PostWhere {
    AND: [PostWhere!]
    NOT: PostWhere
    OR: [PostWhere!]
    author: UserWhere
    authorAggregate: PostAuthorAggregateInput
    authorConnection: PostAuthorConnectionWhere
    authorConnection_NOT: PostAuthorConnectionWhere
    author_NOT: UserWhere
    content: String
    content_CONTAINS: String
    content_ENDS_WITH: String
    content_IN: [String!]
    content_NOT: String @deprecated(reason: \\"Negation filters will be deprecated, use the NOT operator to achieve the same behavior\\")
    content_NOT_CONTAINS: String @deprecated(reason: \\"Negation filters will be deprecated, use the NOT operator to achieve the same behavior\\")
    content_NOT_ENDS_WITH: String @deprecated(reason: \\"Negation filters will be deprecated, use the NOT operator to achieve the same behavior\\")
    content_NOT_IN: [String!] @deprecated(reason: \\"Negation filters will be deprecated, use the NOT operator to achieve the same behavior\\")
    content_NOT_STARTS_WITH: String @deprecated(reason: \\"Negation filters will be deprecated, use the NOT operator to achieve the same behavior\\")
    content_STARTS_WITH: String
}
```

The `authorAggregate` field is totally irrelevant in this scenario.

### Expected behaviour

The field not to be present.

### Version

3.17.0

### Database version

5.4

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by darrellwarde**
# Description

> **Note**
> 
> Please provide a description of the work completed in this PR below

Aggregation fields were being created for relationship fields to single nodes, which serves no functional utility. These will be removed in 4.0.0.

Note: I have no idea why one of the TCK tests changed here!

## Complexity

> **Note**
>
> Please provide an estimated complexity of this PR of either Low, Medium or High

Complexity: Low

# Issue

> **Note**
>
>  Please link to the GitHub issue(s) in which the proposal for this work was discussed
>  
>  To link to multiple issues, use full syntax for each, for example `Closes #1, closes #2, closes #3`

Closes #2969

# Checklist

The following requirements should have been met (depending on the changes in the branch):

- [ ] Documentation has been updated
- [ ] TCK tests have been updated
- [ ] Integration tests have been updated
- [ ] Example applications have been updated
- [ ] New files have copyright header
- [ ] CLA (https://neo4j.com/developer/cla/) has been signed

### CODE REVIEW

**Comment by neo4j-team-graphql** [comment]
# Performance Report

| name | dbHits | old dbHits | time (ms) | old time (ms) | maxRows |
| ---- | ------ | ---------- | --------- | ------------- | ------- |
| 🟩 delete.NestedDeleteInUpdate | 12775 | 16779 | 197 | 154 | 1179 |


🟥 - Performance worsened (dbHits)
🟩 - Performance improved (dbHits)
🟦 - New test
        

<details>
    <summary>Show Full Table</summary>
    
| name | dbHits | old dbHits | time (ms) | old time (ms) | maxRows |
| ---- | ------ | ---------- | --------- | ------------- | ------- |
|  aggregations.TopLevelAggregate | 3403 | 3403 | 33 | 55 | 1134 |
|  aggregations.NestedAggregation | 13514 | 13514 | 90 | 119 | 2174 |
|  aggregations.AggregationWithWhere | 10942 | 10942 | 61 | 72 | 2174 |
|  aggregations.AggregationWhereWithinNestedRelationships | 20101954 | 20101954 | 2968 | 2899 | 2008534 |
|  aggregations.AggregationWhereWithinNestedConnections | 20101954 | 20101954 | 2595 | 2749 | 2008534 |
|  aggregations.NestedCountFromMovieToActors | 8694 | 8694 | 56 | 70 | 2174 |
|  aggregations.NestedCountFromActorsToMovie | 8900 | 8900 | 72 | 80 | 2174 |
|  aggregations.DeeplyNestedCount | 12062294 | 12062298 | 3918 | 4647 | 2008534 |
|  batch-create.BatchCreate | 3600 | 3600 | 112 | 128 | 600 |
|  connect.createAndConnect | 12419 | 12419 | 201 | 339 | 3003 |
|  connections.Connection | 13042 | 13042 | 108 | 166 | 2174 |
|  connections.NestedConnection | 38231 | 38237 | 111 | 122 | 4516 |
|  connections.ConnectionWithSort | 2277 | 2277 | 58 | 80 | 1040 |
|  connections.ConnectionWithSortAndCypher | 14278 | 14278 | 83 | 136 | 2174 |
|  create.SimpleMutation | 6 | 6 | 28 | 40 | 1 |
|  cypher-directive.TopLevelMutationDirective | 1135 | 1135 | 89 | 66 | 1134 |
|  delete.SimpleDelete | 18361 | 18361 | 270 | 291 | 1040 |
| 🟩 delete.NestedDeleteInUpdate | 12775 | 16779 | 197 | 154 | 1179 |
|  query.SimpleQuery | 14120 | 14120 | 171 | 186 | 2174 |
|  query.QueryWhere | 8665 | 8658 | 41 | 38 | 2163 |
|  query.SimpleQueryWithNestedWhere | 8836 | 8829 | 78 | 70 | 2163 |
|  query.Nested | 10084290 | 10084290 | 9757 | 11148 | 2008534 |
|  query.NestedWithFilter | 37172 | 37172 | 163 | 115 | 4511 |
|  query.OrFilterOnRelationships | 37420 | 36607 | 147 | 184 | 1967 |
|  query.OrFilterOnRelationshipsAndNested | 30894 | 30319 | 180 | 314 | 1967 |
|  query.QueryWithNestedIn | 12297 | 12239 | 63 | 86 | 1200 |
|  query.NestedConnectionWhere | 8794 | 8794 | 87 | 69 | 2174 |
|  query.DeeplyNestedConnectionWhere | 8869 | 8898 | 139 | 91 | 2174 |
|  query.DeeplyNestedWithRelationshipFilters | 6251 | 6251 | 146 | 146 | 1134 |
|  query.NestedWithRelationshipSingleFilters | 3021189 | 3021189 | 592 | 591 | 1003942 |
|  query.Fulltext | 96 | 96 | 24 | 34 | 16 |
|  query.FulltextWithNestedQuery | 571 | 571 | 41 | 44 | 84 |
|  sorting.SortMultipleTypes | 2493 | 2493 | 67 | 71 | 1040 |
|  sorting.SortMultipleTypesWithCypherWithCypher | 13519 | 13407 | 83 | 137 | 2174 |
|  sorting.SortOnNestedFields | 13042 | 13042 | 59 | 96 | 2174 |
|  sorting.SortDeeplyNestedFields | 38965 | 38965 | 108 | 134 | 4516 |
|  sorting.SortWithTopLevelCypher | 14120 | 14120 | 53 | 53 | 2174 |
|  unions.SimpleUnionQuery | 321 | 321 | 65 | 73 | 35 |
|  unions.SimpleUnionQueryWithMissingFields | 293 | 293 | 62 | 85 | 35 |
|  unions.NestedUnion | 398553 | 398553 | 285 | 480 | 33033 |
|  unions.NestedUnionWithMissingFields | 372527 | 372527 | 292 | 316 | 33033 |
|  update.NestedUpdate | 2119 | 2119 | 40 | 49 | 1040 |

</details>
        
Old Schema Generation: 28.659s
Schema Generation: 30.607s
Old Subgraph Schema Generation: 41.980s
Subgraph Schema Generation: 41.374s

File: packages/graphql/src/schema/create-relationship-fields/index.ts [hunk 1]
```diff
@@ -571,162 +565,165 @@ function createRelationshipFields({
...
-            }
-
-            const fields = WHERE_AGGREGATION_TYPES.reduce<BaseField[]>((r, t) => {
-                const f = [...nodeOrRelFields.primitiveFields, ...nodeOrRelFields.temporalFields].filter(
-                    (y) => !y.typeMeta.array && y.typeMeta.name === t
-                );
-
-                if (!f.length) {
-                    return r;
+        if (rel.typeMeta.array) {
```
Comment (Liam-Doodson) [inline packages/graphql/src/schema/create-relationship-fields/index.ts:568]: Slightly nit-picky but could this not consolidated with the other  `if (rel.typeMeta.array)` statement that's a few lines below?

**Review by Liam-Doodson — APPROVED** [review]
LGTM! That tck test is weird but the logic hasn't changed luckily. I guess the order cypher builder has assigned the variables changed for some reason!

**Comment by Liam-Doodson** [comment]
The failing TCK tests seem to have the new rel type escaping. I assume that's come frome dev but might be worth looking into because did we not say that was going to be a breaking change?

**Review by MacondoExpress — APPROVED** [review]
The changes look good to me but, to be fair, I'm very confused why the two predicates are translated in a different order in the TCK test!
Btw the cypher-builder variables are different because the query is composed in a different order.

File: packages/graphql/src/schema/create-relationship-fields/index.ts (hunk 1 above)
Comment (darrellwarde) [reply packages/graphql/src/schema/create-relationship-fields/index.ts:577]: I think I've resolved what you were after!

**Comment by darrellwarde** [comment]
> The failing TCK tests seem to have the new rel type escaping. I assume that's come frome dev but might be worth looking into because did we not say that was going to be a breaking change?

Well this is merging into `4.0.0`, so not a huge problem, just a bit weird really!

**Comment by Liam-Doodson** [comment]
> > The failing TCK tests seem to have the new rel type escaping. I assume that's come frome dev but might be worth looking into because did we not say that was going to be a breaking change?
> 
> Well this is merging into `4.0.0`, so not a huge problem, just a bit weird really!

Ah yes ignore me!

**Pull request merged**